flake8_trio.py

```diff
@@ -1146,6 +1147,52 @@ def visit_BoolOp(self, node: ast.BoolOp):
         self.uncheckpointed_statements = worst_case_shortcut
 
 
+_startable_targets: Dict[str, Tuple[str, ...]] = {
+    "trio": ("run_process", "serve_ssl_over_tcp", "serve_tcp", "serve_listeners"),
+    "": ("serve",),
+}
+
+
+class Visitor113(Flake8TrioVisitor):
+    def __init__(self, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+        self.async_function = False
+        self.asynccontextmanager = False
+        self.aenter = False
+
+    def visit_AsyncFunctionDef(self, node: ast.AsyncFunctionDef):
+        outer = self.aenter
+        if any(
+            isinstance(n, ast.Name) and n.id == "asynccontextmanager"
+            for n in node.decorator_list
+        ) or (
+            node.name == "__aenter__"
+            and len(node.args.args) == 1
+            and node.args.args[0].arg == "self"

```

Comment:
We don't need to require that this is `self`; it's a strong convention but not a requirement.

Reply:
ah right. Do you think the `len()` check is worth keeping? Or checking that it's defined inside a class?

Reply:
If in doubt, let's write as little code as possible - I think the len check should stay (the aenter protocol doesn't take args beyond self), but not defined in a class (I've occasionally seen functions manually attached to a class as methods).

Reply:
well, if we're checking `len(args.args)==1` to be paranoid about signature we should also check `args.posonlyargs`, `args.kwonlyargs`, `args.vararg`, `args.kw_defaults` and `args.defaults` are empty...